### ISSUE TITLE
关于代码

### ISSUE THREAD

**Issue by a2260189604-byte** [xulsh8/FlickCD#2]
您好，我阅读了您的论文，并进行了复现，但我每次复现的最佳F1感觉每次都不一样。
第一次
test_Pre=0.8672	 test_Rec:0.7824	 test_OA=0.9204	 test_F1=0.8226	 test_IoU=0.6987	 test_KC=0.7715
第二次。
test_Pre=0.8708	 test_Rec:0.7917	 test_OA=0.9232	 test_F1=0.8294	 test_IoU=0.7085	 test_KC=0.7800
第三次。
test_Pre=0.8577	 test_Rec:0.7963	 test_OA=0.9208	 test_F1=0.8259	 test_IoU=0.7034	 test_KC=0.7747
这是类似这样  感觉每次都有波动。这种不同指标 。我看您这里随机种子基本也是固定的，  是需要下面的注释打开吗？
def set_seed(seed):
    random.seed(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    # torch.backends.cudnn.deterministic = True
    # torch.backends.cudnn.benchmark = False
    # torch.use_deterministic_algorithms(True)
另外您能告诉我您的Pytroch版本吗  谢谢您！

**Comment by xulsh8** [comment]
> 您好，我阅读了您的论文，并进行了复现，但我每次复现的最佳F1感觉每次都不一样。 第一次 test_Pre=0.8672 test_Rec:0.7824 test_OA=0.9204 test_F1=0.8226 test_IoU=0.6987 test_KC=0.7715 第二次。 test_Pre=0.8708 test_Rec:0.7917 test_OA=0.9232 test_F1=0.8294 test_IoU=0.7085 test_KC=0.7800 第三次。 test_Pre=0.8577 test_Rec:0.7963 test_OA=0.9208 test_F1=0.8259 test_IoU=0.7034 test_KC=0.7747 这是类似这样 感觉每次都有波动。这种不同指标 。我看您这里随机种子基本也是固定的， 是需要下面的注释打开吗？ def set_seed(seed): random.seed(seed) np.random.seed(seed) torch.manual_seed(seed) torch.cuda.manual_seed(seed) # torch.backends.cudnn.deterministic = True # torch.backends.cudnn.benchmark = False # torch.use_deterministic_algorithms(True) 另外您能告诉我您的Pytroch版本吗 谢谢您！

Thank you very much for your interest in our work.
Regarding the reproducibility of the experimental results, You only need to uncomment `torch.use_deterministic_algorithms(True)` and the line `export CUBLAS_WORKSPACE_CONFIG=:4096:8` in **trainAndtest.sh**. This will ensure that each experiment is determined solely by the random seed. As for the PyTorch version, we used `2.7.1+cu126`.

**Comment by a2260189604-byte** [comment]
> > 您好，我阅读了您的论文，并进行了复现，但我每次复现的最佳F1感觉每次都不一样。第一次 test_Pre=0.8672 test_Rec：0.7824 test_OA=0.9204 test_F1=0.8226 test_IoU=0.6987 test_KC=0.7715 第二次。test_Pre=0.8708 test_Rec：0.7917 test_OA=0.9232 test_F1=0.8294 test_IoU=0.7085 test_KC=0.7800 第三次。test_Pre=0.8577 test_Rec：0.7963 test_OA=0.9208 test_F1=0.8259 test_IoU=0.7034 test_KC=0.7747 这是类似这样 感觉每次都有波动。这种不同指标 。我看您这里随机种子基本也是固定的，是需要下面的注释打开吗？ def set_seed（seed）： random.seed（seed） np.random.seed（seed） torch.manual_seed（seed） torch.cuda.manual_seed（seed） # torch.backends.cudnn.deterministic = True # torch.backends.cudnn.benchmark = False # torch.use_deterministic_algorithms（True） 另外您能告诉我您的Pytroch版本吗 谢谢您！
> 
> 非常感谢您对我们工作的关注。关于实验结果的可重复性，您只需取消注释和 **trainAndtest.sh** 中的行即可。这将确保每个实验仅由随机种子决定。至于 PyTorch 版本，我们使用了 .`torch.use_deterministic_algorithms(True)``export CUBLAS_WORKSPACE_CONFIG=:4096:8``2.7.1+cu126`

您好非常感谢您的回答，我是 2.7.1+CU118 当我开启了 torch.use_deterministic_algorithms(True)``export CUBLAS_WORKSPACE_CONFIG=:4096:8  经过了多次实验结果复现发现结果始终为 test_Pre=0.8684	 test_Rec:0.7894 test_OA=0.9221	 test_F1=0.8270 test_IoU=0.7050est_KC=0.7769   结果始终为这个，这解决了我的疑惑
但我不使用torch.use_deterministic_algorithms(True)的时候，只使用export CUBLAS_WORKSPACE_CONFIG=:4096:8  
实验结果test_Pre=0.8592 test_Rec:0.8064 test_OA=0.9232 test_F1=0.8319 test_IoU=0.7123	 test_KC=0.7822  
多次训练后平均F1基本达到了0.8305左右上  也就是跟开启确定性算法相比 提升了0.35个点，我感觉.torch.use_deterministic_algorithms(True)export CUBLAS_WORKSPACE_CONFIG=:4096:8一起使用降低了模型的精度，就是我有个
疑问，您训练最佳模型的时候，是怎么样设置的呢？非常感谢您！

**Comment by xulsh8** [comment]
> > > 您好，我阅读了您的论文，并进行了复现，但我每次复现的最佳F1感觉每次都不一样。第一次 test_Pre=0.8672 test_Rec：0.7824 test_OA=0.9204 test_F1=0.8226 test_IoU=0.6987 test_KC=0.7715 第二次。test_Pre=0.8708 test_Rec：0.7917 test_OA=0.9232 test_F1=0.8294 test_IoU=0.7085 test_KC=0.7800 第三次。test_Pre=0.8577 test_Rec：0.7963 test_OA=0.9208 test_F1=0.8259 test_IoU=0.7034 test_KC=0.7747 这是类似这样 感觉每次都有波动。这种不同指标 。我看您这里随机种子基本也是固定的，是需要下面的注释打开吗？ def set_seed（seed）： random.seed（seed） np.random.seed（seed） torch.manual_seed（seed） torch.cuda.manual_seed（seed） # torch.backends.cudnn.deterministic = True # torch.backends.cudnn.benchmark = False # torch.use_deterministic_algorithms（True） 另外您能告诉我您的Pytroch版本吗 谢谢您！
> > 
> > 
> > 非常感谢您对我们工作的关注。关于实验结果的可重复性，您只需取消注释和 **trainAndtest.sh** 中的行即可。这将确保每个实验仅由随机种子决定。至于 PyTorch 版本，我们使用了 .`` torch.use_deterministic_algorithms(True)``export CUBLAS_WORKSPACE_CONFIG=:4096:8``2.7.1+cu126 ``
> 
> 您好非常感谢您的回答，我是 2.7.1+CU118 当我开启了 torch.use_deterministic_algorithms(True)``export CUBLAS_WORKSPACE_CONFIG=:4096:8 经过了多次实验结果复现发现结果始终为 test_Pre=0.8684 test_Rec:0.7894 test_OA=0.9221 test_F1=0.8270 test_IoU=0.7050est_KC=0.7769 结果始终为这个，这解决了我的疑惑 但我不使用torch.use_deterministic_algorithms(True)的时候，只使用export CUBLAS_WORKSPACE_CONFIG=:4096:8 实验结果test_Pre=0.8592 test_Rec:0.8064 test_OA=0.9232 test_F1=0.8319 test_IoU=0.7123 test_KC=0.7822 多次训练后平均F1基本达到了0.8305左右上 也就是跟开启确定性算法相比 提升了0.35个点，我感觉.torch.use_deterministic_algorithms(True)export CUBLAS_WORKSPACE_CONFIG=:4096:8一起使用降低了模型的精度，就是我有个 疑问，您训练最佳模型的时候，是怎么样设置的呢？非常感谢您！

In fact, `torch.use_deterministic_algorithms(True)` and `export CUBLAS_WORKSPACE_CONFIG=:4096:8` are only used to ensure that the training results remain the same when a fixed random seed is set. However, even with the same seed, training results may still vary across different GPUs.
The results reported in our paper were obtained by trying multiple random seeds and selecting the best outcome, so there may be some fluctuations when reproducing the training process. We have also uploaded the model weights corresponding to the results shown in the paper, which we hope can help clarify your concerns.
